odoo/fields.py

```diff
@@ -538,6 +535,16 @@ def setup(self, model):
                 self.setup_related(model)
             else:
                 self.setup_nonrelated(model)
+
+            if self.required not in (True, False):
```

Comment:
`not in` check equality not the identity. Then `(1 not in (True, False))` is False